File: 프로그래머스/JavaScript/0단계/각도기.js
// 내 풀이
function solution(angle) {
    return  ((0 < angle) && (angle< 90)) ? 1 : (angle == 90) ? 2 :((90 < angle) && (angle < 180)) ? 3 : 4;
}

// 내 풀이를 개선한 풀이
function solution(angle) {
    return angle < 90 ? 1 : angle === 90 ? 2 : angle < 180 ? 3 : 4;
}


// 다른사람 풀이
function solution(angle) {
    return [0, 90, 91, 180].filter(x => angle>=x).length;
}

File: 프로그래머스/Python/0단계/각도기.py
# 각에서 0도 초과 90도 미만은 예각, 90도는 직각, 90도 초과 180도 미만은 둔각 180도는 평각으로 분류합니다.
# 각 angle이 매개변수로 주어질 때 예각일 때 1, 직각일 때 2, 둔각일 때 3, 평각일 때 4를 return하도록 solution 함수를 완성하는 문제.


# 내 풀이

def solution(angle):
    if angle < 90:
        return 1
    elif angle == 90:
        return 2
    elif 90<angle and angle<180:
        return 3
    else:
        return 4
      
      
# 다른사람 풀이

def solution(angle):
    answer = (angle // 90) * 2 + (angle % 90 > 0) * 1
    return answer
